ENH: stats: add Zipfian (different from Zipf/zeta) distribution

mdhaber opened the pull request:
#### Reference issue
Closes gh-11523

#### What does this implement/fix?
gh-11523 notes that `scipy.stats.zipf` implements the zeta distribution. The zeta distribution is sometime times called the Zipf distribution, but Zipf/Zipfian is actually the name of distinct (more general) distribution.  This PR adds the Zipfian distribution and clarifies the difference between the two in the documentation. As an additional step to address the issue, we might want to add `zeta` as an alias for the existing `zipf` and possibly doc-deprecate `zipf`.

#### Additional information
I've tried to follow the [guide](http://scipy.github.io/devdocs/dev/contributor/adding_new.html#adding-new) to adding a new distribution, but please let me know what I need to change.

mdhaber commented on doc/source/tutorial/stats/discrete_zipfian.rst:
```diff
@@ -0,0 +1,38 @@
+Zipfian Distribution
+========================
+
+A random variable has the Zipfian distribution with parameters
```
Followed the language of the Zipf distribution.

mdhaber commented on doc/source/tutorial/stats/discrete_zipfian.rst:
```diff
@@ -0,0 +1,38 @@
+========================
+
+A random variable has the Zipfian distribution with parameters
+:math:`s \ge 0` and :math:`N \in {1, 2, 3, \dots}` if its probability
```
Chose these to match the parameter names in Wikipedia.

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1018,13 +1018,21 @@ class zipf_gen(rv_discrete):
     Riemann zeta function (`scipy.special.zeta`)
 
+    The Zipf distribution is also known as the zeta distribution, which is
+    a special case of the Zipfian distribution (`zipfian`).
```
Suggestions for more specific language than "special case"?

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1018,13 +1018,21 @@ class zipf_gen(rv_discrete):
+    https://en.wikipedia.org/wiki/Zeta_distribution
+
     %(example)s
 
```
Shall I add an example that shows the equivalence as `N -> oo`?

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,57 @@ def _munp(self, n, a):
+        return (a >= 0) & (n == n.astype(int))
+
+    def _pmf(self, k, a, n):
+        Pk = 1.0 / _gen_harmonic(n, a) / k**a
```
Shall I put this on one line? I was just matching zipf.

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,57 @@ def _munp(self, n, a):
+    :math:`H_{n,a}` is the :math:`n^{\mbox{th}}` generalized harmonic number
+    of order :math:`a`.
+
+    The Zipfian distribution converges to the Zipf (zeta) distribution as
```
More appropriate language than "converges"?

ev-br replied:
Maybe "reduces to"?

mdhaber replied:
Perfect; I'll change today.

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,57 @@ def _munp(self, n, a):
+
+    .. math::
+
+        f(k, a, n) = \frac{1}{H_{n,a} k^a}
```
Used `a` to match `zipf`.

mdhaber commented on scipy/stats/tests/test_discrete_distns.py:
```diff
@@ -96,3 +98,12 @@ def test_betabinom_bernoulli():
+    N = 10000000
+    k = np.arange(1, 21)
+    assert_allclose(zipfian.pmf(k, a, N), zipf.pmf(k, a))
+    assert_allclose(zipfian.cdf(k, a, N), zipf.cdf(k, a))
```
I can add tests to test against some other reference.

mdhaber replied:
zipfUC from R's VGAM package seems innacurate for low `a`, `N`. Trying this at [rdrr.io](https://rdrr.io/cran/VGAM/man/zipfUC.html):
```
library(VGAM)
print(dzipf(0:21, shape = 1.01, N = 20))
print(sum(dzipf(0:21, shape = 1.01, N = 20)))
print(pzipf(20, shape = 1.01, N = 20))
```
The first line prints the pmf, the second is the sum of the pmf, and the third is the cdf. The sum of the pmf values is not 1.

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
+
+class zipfian_gen(rv_discrete):
+    r"""A Zipfian discrete random variable.
+
```
Add see also?

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
+        mu2 = mu2n / mu2d
+        g1 = (Hna3/Hna - 3*Hna1*Hna2/Hna**2 + 2*Hna1**3/Hna**3)/mu2**(3/2)
+        g2 = (Hna**3*Hna4 - 4*Hna**2*Hna1*Hna3 + 6*Hna*Hna1**2*Hna2
+              - 3*Hna1**4) / mu2n**2 - 3
```
```suggestion
              - 3*Hna1**4) / mu2n**2 - 3
```
Why is subtracting 3 needed? I see this in `dlaplace_gen` and `betabinom_gen`, and that's exactly what was needed here.

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
+    def _sf(self, k, a, n):
+        k += 1  # # to match SciPy convention
+        # see http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Zipf.pdf
+        # implemented exactly as shown; please suggest improvements
```
Does this `sf` have the potential to be more precise than `1-cdf`? Here and in `_stats` below, implementation follows the reference directly; suggestions for improvement to precision welcome.

ev-br commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
+
+    """
+    def _argcheck(self, a, n):
+        return (a >= 0) & (np.asarray(n) == np.asarray(n, dtype=int))
```
Asarrays not needed? IIRC all is converted to arrays when _argcheck is called

mdhaber replied:
Yes I meant to check. Thanks for the reminder.

mdhaber replied:
Apparently not:
```
return (a >= 0) & (n == n.astype(int))
E   AttributeError: 'int' object has no attribute 'astype'
```

mdhaber replied:
This happens in the `moment` method. I think it is being fixed in gh-12197.

rlucas7 replied:
@mdhaber should this PR wait for gh-12197 to be merged then?

mdhaber replied:
I don't think so. It's not really hurting anything to ensure it's an array.

ev-br commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
+        return  _gen_harmonic(k, a)/_gen_harmonic(n, a)
+
+    def _sf(self, k, a, n):
+        k += 1  # # to match SciPy convention
```
Being paranoid: can you check this does not modify input inplace? Esp if k is already an array

mdhaber replied:
Yes I forgot about that for arrays. I'll change it.

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1066,89 @@ def _munp(self, n, a):
+    """Generalized harmonic number"""
+    # See https://en.wikipedia.org/wiki/Harmonic_number; search for "hurwitz"
+    # todo - make this work for a < 1 (can't use zeta)
+    return zeta(a, 1) - zeta(a, n+1)
```
How should I implement this for a <=1 ? I was thinking a naive (just summing the harmonic series) in a ufunc. I could try to do something fancy from within Python that saves computation (e.g. for each distinct `a`, generate the harmonic series up to the maximum of the `n`s, perform a cumulative sum for the `n`s that are needed) but I'm not sure it's worth the trouble.

mdhaber commented:
I anticipate two test failures.  Both are ultimately caused by trouble with `expect`:
```
import numpy as np
from scipy.stats import zipf, zipfian
a, n = 3.5, 10
print(zipfian.moment(0, a, n))
print(zipfian.expect(lambda x: 1, (a, n)))
```
These are supposed to both be 1, but we get:
```
1.0
1.0009914161108657
```

The second is from difficulty calculating the entropy:
```
print(zipfian.entropy(a, n))
```
and we get a warning:
```
RuntimeWarning: expect(): sum did not converge
```
Note that the existing `zipf` distribution suffers from the same problem, it's just not exposed because the "sane parameter" was chosen as `a=6.5`, for which neither distribution has any trouble. There is a formula for the entropy on Wikipedia, but it is not very stable.

mdhaber commented on scipy/stats/tests/test_discrete_distns.py:
```diff
@@ -112,3 +114,81 @@ def test_skellam_gh11474():
+        assert_allclose(zipfian.stats(alt1, N, moments='msvk'),
+                        zipfian.stats(agt1, N, moments='msvk'), rtol=2e-7)
+
+    def test_zipfian_R(self):
```
I should remove this test against R because the R implementation is wrong. For example, run this code at [rdrr.io](https://rdrr.io/cran/VGAM/man/zipfUC.html).
```R
library(VGAM)
options(digits=16)
k <- 3
a <- 1.2589254117941673
n <- 12
pmf <- dzipf(k, N = n, shape = a)
cdf <- pzipf(k, N = n, shape = a)
print(pmf)
print(cdf)
```
For `n <= 12`, the two implementations match to machine precision.
Increase `n` to 13, and the error is ~0.001. You can tell R is wrong because if you set:
```R
k <- 0:14
```
and sum the pmf values, they don't add to 1.

mckib2 replied:
So I assume the goal of this test is to provide reasonable assurance that this SciPy implementation is doing a reasonable thing compared to another independent implementation, in this case R.  Is Wolfram's `ZipfDistribution` the correct distribution? If not, might be okay to omit this test if we're testing against bad values

FYI, appears that R [is using](https://github.com/cran/VGAM/blob/ad275cd8262668ccf792601c74c008f5014285d9/R/family.aunivariate.R#L4815) what you call below the "naive" method

mdhaber replied:
Yes, the idea is just to compare against an existing implementation.

>  Is Wolfram's ZipfDistribution the correct distribution?

No, that's the zeta distribution. The confusion between the two distributions (gh-11523) is why I created this PR in the first place, actually : )  So another thing I was thinking we might want to do is doc-deprecate `zipf` and give it an alias `zeta`.

mckib2 replied:
Nope, I don't think it's license compatible, it's GPL 2 or 3

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1068,118 @@ def _munp(self, n, a):
+
+    """
+    def _argcheck(self, a, n):
+        return (a >= 0) & (n > 0)
```
I had a check for n being an integer in here, but I removed it because I don't see it for any other discrete distribution which is supposed to have an integer shape. It generalizes in a reasonable way for a > 1, but not for a < 1. Should I add in the integer check?

mckib2 replied:
I think an integer check makes sense if something breaks when `n` isn't an integer.  If it still "works" if `n` isn't an integer, then if it seems like it would be consistent with the existing stats distributions to not check I would follow precedence

mdhaber replied:
OK adding integer check back.

mdhaber commented on doc/source/tutorial/stats/discrete_zipfian.rst:
```diff
@@ -0,0 +1,41 @@
+
+    H_{N, s}=\sum_{n=1}^{N}\frac{1}{n^{s}}
+
+is the :math:`N\textsuperscript{th}` generalized harmonic number of order
```
```suggestion
is the :math:`N`:sup:`th` generalized harmonic number of order
```

mdhaber commented on doc/source/tutorial/stats/discrete_zipfian.rst:
```diff
@@ -0,0 +1,41 @@
+.. math::
+   :nowrap:
+
+    \begin{eqnarray*} p\left(k; s, N \right) & = & \frac{1}{H_{N, s}k^{s}}\quad k\geq1\end{eqnarray*}
```
```suggestion
    \begin{eqnarray*} p\left(k; s, N \right) & = & \frac{1}{H_{N, s}k^{s}}\quad k \in \{1, 2, \dots, n-1, n\} \end{eqnarray*}
```

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1068,118 @@ def _munp(self, n, a):
+    def _sf(self, k, a, n):
+        k = k + 1  # # to match SciPy convention
+        # see http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Zipf.pdf
+        # implemented exactly as shown; please suggest improvements
```
```suggestion
```

mdhaber commented on scipy/stats/tests/test_discrete_distns.py:
```diff
@@ -112,3 +114,81 @@ def test_skellam_gh11474():
+
+    @pytest.mark.parametrize("a, n", [(1.01, 10), (1.56712977, 70),
+                                      (3.681, 33), (5.492, 66)])
+    def test_zipfian_naive(self, a, n):
```
I think this is a fine test if I adjust the values of a it tests to be logarithmically distributed between some small number and some large number.

mdhaber replied:
Expanded these tests.

mdhaber commented on scipy/stats/tests/test_discrete_distns.py:
```diff
@@ -112,3 +114,81 @@ def test_skellam_gh11474():
+    @pytest.mark.parametrize("a, n", [(1.01, 10), (1.56712977, 70),
+                                      (3.681, 33), (5.492, 66)])
+    def test_zipfian_naive(self, a, n):
+        # test against bare-bones implementation of zipfian
```
```suggestion
        # test against bare-bones implementation of Zipfian
```

rlucas7 commented:
The 3 comes from a kurtosis of a normal/Gaussian. This is sometimes called kurtosis excess, the excess w.r.t. a Gaussian. 

For better or worse, many things in statistics are relative to a Gaussian. 

-L

> On Dec 6, 2020, at 6:03 AM, Matt Haberland <notifications@github.com> wrote:
> 
> ﻿
> @mdhaber commented on this pull request.
> 
> In scipy/stats/_discrete_distns.py:
> 
> > +                / (k**a*_gen_harmonic(n, a)))
> +
> +    def _stats(self, a, n):
> +        # see # see http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Zipf.pdf
> +        Hna = _gen_harmonic(n, a)
> +        Hna1 = _gen_harmonic(n, a-1)
> +        Hna2 = _gen_harmonic(n, a-2)
> +        Hna3 = _gen_harmonic(n, a-3)
> +        Hna4 = _gen_harmonic(n, a-4)
> +        mu1 = Hna1/Hna
> +        mu2n = (Hna2*Hna - Hna1**2)
> +        mu2d = Hna**2
> +        mu2 = mu2n / mu2d
> +        g1 = (Hna3/Hna - 3*Hna1*Hna2/Hna**2 + 2*Hna1**3/Hna**3)/mu2**(3/2)
> +        g2 = (Hna**3*Hna4 - 4*Hna**2*Hna1*Hna3 + 6*Hna*Hna1**2*Hna2
> +              - 3*Hna1**4) / mu2n**2 - 3
> Why is subtracting 3 needed? I see this in dlaplace_gen and betabinom_gen, and that's exactly what was needed here to match what I was expecting.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

rlucas7 commented:
> On Dec 28, 2020, at 2:13 AM, Matt Haberland <notifications@github.com> wrote:
> 
> ﻿
> @mdhaber commented on this pull request.
> 
> In scipy/stats/_discrete_distns.py:
> 
> > +    .. [1] "Zipf's Law", Wikipedia, https://en.wikipedia.org/wiki/Zipf's_law
> +    .. [2] Larry Leemis, "Zipf Distribution", Univariate Distribution
> +           Relationships. http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Zipf.pdf
> +
> +    %(example)s
> +
> +    Confirm that `zipfian` reduces to `zipf` for large `n`, `a > 1`.
> +
> +    >>> from scipy.stats import zipf
> +    >>> k = np.arange(11)
> +    >>> np.allclose(zipfian.pmf(k, a=3.5, n=10000000), zipf.pmf(k, a=3.5))
> +    True
> +
> +    """
> +    def _argcheck(self, a, n):
> +        return (a >= 0) & (n > 0)
> I had a check for n being an integer in here, but I removed it because I don't see it for any other discrete distribution which is supposed to have an integer shape. It generalizes in a reasonable way for a > 1, but not for a < 1. Should I add in the integer check?
> 
I see why that’s a reasonable thing to do but for consistency with the rest of codebase I’d omit, unless people have strong opinions to include. 


> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

rlucas7 commented:
@mdhaber this looks mostly ready, is there anything else you wanted to change? 

If not I can take a closer look over the next few days. 

Sincerely, 

-Lucas Roberts 

> On Jan 20, 2021, at 8:35 PM, rlucas7@vt.edu wrote:
> 
> ﻿The 3 comes from a kurtosis of a normal/Gaussian. This is sometimes called kurtosis excess, the excess w.r.t. a Gaussian. 
> 
> For better or worse, many things in statistics are relative to a Gaussian. 
> 
> -L
> 
>>> On Dec 6, 2020, at 6:03 AM, Matt Haberland <notifications@github.com> wrote:
>>> 
>> ﻿
>> @mdhaber commented on this pull request.
>> 
>> In scipy/stats/_discrete_distns.py:
>> 
>> > +                / (k**a*_gen_harmonic(n, a)))
>> +
>> +    def _stats(self, a, n):
>> +        # see # see http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Zipf.pdf
>> +        Hna = _gen_harmonic(n, a)
>> +        Hna1 = _gen_harmonic(n, a-1)
>> +        Hna2 = _gen_harmonic(n, a-2)
>> +        Hna3 = _gen_harmonic(n, a-3)
>> +        Hna4 = _gen_harmonic(n, a-4)
>> +        mu1 = Hna1/Hna
>> +        mu2n = (Hna2*Hna - Hna1**2)
>> +        mu2d = Hna**2
>> +        mu2 = mu2n / mu2d
>> +        g1 = (Hna3/Hna - 3*Hna1*Hna2/Hna**2 + 2*Hna1**3/Hna**3)/mu2**(3/2)
>> +        g2 = (Hna**3*Hna4 - 4*Hna**2*Hna1*Hna3 + 6*Hna*Hna1**2*Hna2
>> +              - 3*Hna1**4) / mu2n**2 - 3
>> Why is subtracting 3 needed? I see this in dlaplace_gen and betabinom_gen, and that's exactly what was needed here to match what I was expecting.
>> 
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub, or unsubscribe.

rlucas7 commented on scipy/stats/tests/test_discrete_distns.py:
```diff
@@ -96,3 +98,15 @@ def test_betabinom_bernoulli():
     assert_almost_equal(p, expected)
+
+
+def test_zipfian():
```
should this test be marked slow? I'm assuming it will be.

mdhaber replied:
In the version of the code you commented on, `test_zipfian` is equivalent to the test now called `test_zipfian_asymptotic`. 

For `a` greater than 1, the generalized harmonic numbers are not calculated by taking the sum of a series; they use the zeta function, so the magnitude of `n` doesn't really affect the speed. The test is for `k = np.arange(1, 21)`, and it doesn't take too long to do 21 of these calculations.

This test is slow:
```
7.94s call     stats/tests/test_discrete_basic.py::test_discrete_basic[zipfian-arg17-True]
```
so I should check skip whatever is taking a long time there - maybe `rvs`.

rlucas7 replied:
these are comments on outdated diff, marking resolved

mdhaber replied:
For now I've effectively marked the `zipfian` `test_discrete_basic` as slow, so it only runs when test mode is full.

rlucas7 reviewed (commented):
@mdhaber I left two (very minor) comments and it looks like there might be 1 or two open changes from @ev-br review. 

Only substantive comment I had was if you wanted to put the rejection sampler in for this distribution?  I imaging the sequential search can be slow for certain parameter values (closer to 1). The algorithm is fairly short so it wouldn't be much more than: 

```
import numpy as np

def _rvs(a=2):
    # assumes a > 1
    b = 2 ** (a - 1)
    rs = np.random.RandomState()
    while True:
        u, v = rs.uniform(0,1,2)
        x = np.floor(u ** (- 1/ (a-1) ))
        t = (1+1/x) ** (a - 1)
        if v * x * ( (t - 1) / (b - 1)) <= t / b :
            return x
```

but you might want to clean up a bit, e.g. for larger values of `a` I'm not sure if it still makes sense to always use rejection-it might I just didn't test that out. 

Of course this assumes I understood the distinction between *zipf* and *zeta* and it's possible I misunderstood.

rlucas7 commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1064,6 +1083,116 @@ def _munp(self, n, a):
+
+    %(example)s
+
+    Confirm that `zipfian` reduces to `zipf` for large `n`, `a > 1`.
```
Ah I see here the distinction. The rejection sampler I wrote in comment is for `zipf` not `zipfian`-I'll edit.

mdhaber commented on scipy/stats/_discrete_distns.py:
```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
+        return  _gen_harmonic(k, a)/_gen_harmonic(n, a)
+
+    def _sf(self, k, a, n):
+        k += 1  # # to match SciPy convention
```
```suggestion
        k = k + 1  # # to match SciPy convention
```

mdhaber commented:
> @mdhaber I left two (very minor) comments and it looks like there might be 1 or two open changes from @ev-br review.

It looks like you commented on an outdated commit. I believe @ev-br's review has been addressed, but I don't click resolve on reviewer comments. When I review PRs, I prefer to to be the one to click resolve so I can keep track of what changed in response and verify that the change sufficiently addresses the comment.

> so we can omit from this PR unless you feel strongly to include.

I'd prefer to omit. That can be an enhancement.

> I see why that’s a reasonable thing to do but for consistency with the rest of codebase I’d omit, unless people have strong opinions to include.  EDIT: I see from the conversion (now that I'm logged in on laptop) that something broke and keeping int check makes sense to fail fast.

Well, I don't know if there's an error for non-integers, but when `a < 1` we sum `n` terms of a sequence, and I'm not sure how to sum a fraction of a term, so the result would be wrong.

rlucas7 commented:
> > @mdhaber I left two (very minor) comments and it looks like there might be 1 or two open changes from @ev-br review.
> 
> It looks like you commented on an outdated commit. 

Yeah, sometimes github doesn't show me the latest commit diffs, I'm probably doing something wrong. 

>I believe @ev-br's review has been addressed, but I don't click resolve on reviewer comments. When I review PRs, I prefer to to be the one to click resolve so I can keep track of what changed in response and verify that the change sufficiently addresses the comment.

Yeah I agree, comment does look addressed, maybe @ev-br can comment/close that one? 

> > so we can omit from this PR unless you feel strongly to include.
> 
> I'd prefer to omit. That can be an enhancement.

That's fine. I can open a PR once this is merged, it's pretty clear to me the change and cases.
 
> > I see why that’s a reasonable thing to do but for consistency with the rest of codebase I’d omit, unless people have strong opinions to include.  EDIT: I see from the conversion (now that I'm logged in on laptop) that something broke and keeping int check makes sense to fail fast.
> 
> Well, I don't know if there's an error for non-integers, but when `a < 1` we sum `n` terms of a sequence, and I'm not sure how to sum a fraction of a term, so the result would be wrong.

Right, `n` should be int here, otherwise things won't be correct.

AFAICS the PR looks ready to merge, let's give @ev-br a week or so to comment. It's easy to miss messages on github though so if no comments for changes within a week or so feel free to ping me and I can merge this one.

mdhaber commented:
Thanks very much @rlucas7! If you could check one thing before then - I just pushed a commit that marks certain cases of `test_discrete_basic` as slow, so they only run when test mode is full.  This is the same way it is done in `test_continuous_basic.py`. I went ahead and marked both `zipfian` and `nhypergeom`, since that was pretty slow, too.

rlucas7 commented on scipy/stats/tests/test_discrete_basic.py:
```diff
@@ -14,31 +14,22 @@
     for distname, arg in distdiscrete:
-        yield distname, arg, distname not in seen
+        if distname in distslow:
+            yield pytest.param(distname, arg, distname, marks=pytest.mark.slow)
```
I was wondering how the tested get marked for pytest to still pickup the slow marker, it's in this line. LGTM.

rlucas7 reviewed (approved):
@mdhaber LGTM. 
I'll leave it open until this Sunday Jan 31st, if no further comments by morning EST I'll merge.

rlucas7 commented:
Everything green and no further comments so squashing and merging. 
Thanks @mdhaber and @ev-br for the review. 

@mdhaber I'll add an entry to the release notes wiki-please update if anything needs adding/correcting in the release notes entry

mdhaber commented:
Thanks @rlucas7 @ev-br!
The original issue noted confusion about the `zipf` distribution, which could mean `zipfian` but is also known as `zeta`. There is less chance of confusion between `zeta` and `zipfian`. Should I suggest on the mailing list that we add `zeta` as an alias for the existing `zipf` and deprecate `zipf`?

rlucas7 commented:
@mdhaber

I think that might be a good idea but I’m not sure what others think, I was confused by the difference initially but not necessarily because of naming here in scipy. Rather that the naming seems to be inconsistent acrosss papers and resources I have. 

Mailing list would be good to get others thoughts so we come up with better naming-naming is difficult.

-L

> On Jan 31, 2021, at 11:51 PM, Matt Haberland <notifications@github.com> wrote:
> 
> ﻿
> Thanks @rlucas7 @ev-br!
> The original issue noted confusion about the zipf distribution, which could mean zipfian but is also known as zeta. There is less chance of confusion between zeta and zipfian. Should I suggest on the mailing list that we add zeta as an alias for the existing zipf and deprecate zipf?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

mdhaber commented:
Thanks. I actually don't think I'll bring it up. I still think it's a good idea, there are just higher priority things (even if this would be easy/fast).
